test.js

```diff
@@ -134,6 +142,64 @@ test('execa.shellSync() includes stdout and stderr in errors for improved debugg
 	t.throws(() => execa.shellSync('node fixtures/error-message.js'), {message: STDERR_STDOUT_REGEXP, code: 1});
 });
 
+test('execa() with .kill after it with SIGKILL should kill cleanly', async t => {
+	const subprocess = execa('node', ['fixtures/sub-process-no-killable'], {
+		stdio: ['ipc']
+	});
+
+	await waitForProcessLaunch(subprocess);
+	subprocess.kill('SIGKILL');
+
+	const {signal} = await t.throwsAsync(subprocess);
+	t.is(signal, 'SIGKILL');
+});
+
+test('execa() with .kill after it with SIGTERM should not kill (no retry)', async t => {
+	const subprocess = execa('node', ['fixtures/sub-process-no-killable'], {
+		stdio: ['ipc']
+	});
+
+	await waitForProcessLaunch(subprocess);
+
+	subprocess.kill('SIGTERM', {
+		retry: false,
+		retryAfter: 50
+	});
+
+	t.true(isRunning(subprocess.pid));
+	subprocess.kill('SIGKILL');
+});
+
+test('execa() with .kill after it with SIGTERM should kill after 50 ms with SIGKILL', async t => {
+	const subprocess = execa('node', ['fixtures/sub-process-no-killable'], {
+		stdio: ['ipc']
+	});
+
+	await waitForProcessLaunch(subprocess);
+
+	subprocess.kill('SIGTERM', {
+		retryAfter: 50
+	});
+
+	const {signal} = await t.throwsAsync(subprocess);
+	t.is(signal, 'SIGKILL');
+});
+
+test('execa() with .kill after it with nothing (undefined) should kill after 50 ms with SIGKILL', async t => {
```

Comment:
This test is identical to the one above, i.e. I think you meant to do `kill()` with no arguments

Reply:
Argh. Fixed: 79c3411a8c42db38af115ebf46c791ce78f1361b